src/routes/safe/components/Apps/components/AddAppForm/AppUrl.tsx

```diff
@@ -71,7 +80,9 @@ const AppUrl = ({ appList }: { appList: SafeApp[] }): React.ReactElement => {
   const { visited } = useFormState({ subscription: { visited: true } })
 
   // trick to prevent having the field validated by default. Not sure why this happens in this form
-  const validate = !visited?.appUrl ? undefined : composeValidators(required, validateUrl, uniqueApp(appList))
+  const validate = !visited?.appUrl
+    ? undefined
+    : composeValidators(required, validateUrl, validateManifest, uniqueApp(appList))
```

Comment:
`uniqueApp` should probably go before `validateManifest`.
So that it can check if the URL has been already added before fetching it.